OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -605,8 +605,10 @@ public void ResolveOrder(Actor self, Order order)
 				{
 					if (IsPlane)
 						self.QueueActivity(new ReturnToBase(self, Info.AbortOnResupply));
-					else
+					else if (!Info.RepairBuildings.Contains(order.TargetActor.Info.Name))
```

Comment:
Do you want to move that check and keep the behaviour (do nothing) or you mean to do that rewrite? This PR provides simple fix for current bug until it is rewritten. I can move that small bugfix, rewrite wasn't intended with this PR. In such case, let's close it.